File: backend/app/schemas/news.py
from datetime import datetime

from pydantic import BaseModel


class NewsArticle(BaseModel):
    title: str
    source: str
    url: str
    published_at: datetime
    description: str | None = None


class SignificantMoveEvent(BaseModel):
    date: str                          # YYYY-MM-DD; from OHLCV bar date
    percent_change: float              # rounded to 2 dp
    direction: str                     # "up" | "down"
    close_price: float                 # rounded to 2 dp
    news_available: bool               # False when date is outside the news provider window
    news_unavailable_reason: str | None = None
    headlines: list[NewsArticle]       # empty when news_available=False


class EventsResponse(BaseModel):
    ticker: str
    time_range: str
    threshold_pct: float
    events: list[SignificantMoveEvent]

File: frontend/src/types/news.ts
export interface NewsArticle {
  title: string;
  source: string;
  url: string;
  published_at: string; // ISO datetime string
  description: string | null;
}

export interface SignificantMoveEvent {
  date: string;               // YYYY-MM-DD
  percent_change: number;
  direction: "up" | "down";
  close_price: number;
  news_available: boolean;
  news_unavailable_reason: string | null;
  headlines: NewsArticle[];
}

export interface EventsResponse {
  ticker: string;
  time_range: string;
  threshold_pct: number;
  events: SignificantMoveEvent[];
}
